ambry-commons/src/main/java/com.github.ambry.commons/ReadableStreamChannelInputStream.java

```diff
@@ -55,7 +55,7 @@ public int read()
     int data = -1;
```

Comment:
`available()` cannot return -1 since the constructor initializes it correctly and `reportBytesRead()` makes sure it doesn't go under 0. 

Your other point is correct and I've now modified the function to return whatever is remaining in the current chunk.